files: improve error for uncompressed Helm .tar archives

bashSunny101 opened the pull request:
**PR description**
Summary
- Detects when a user provides an uncompressed `.tar` (not a gzipped chart) to Helm chart parsing and returns a clear, actionable error telling them to use `.tgz` / `.tar.gz` (or `helm package`).

What changed
- Add `ErrUncompressedTarProvided` to surface a helpful message and remedies when Helm's gzip reader fails.
- Update `ParseFileAsHelmChart` (`files/identification.go`) to detect the `"gzip: invalid header"` failure for `.tar` and return the new error.
- Add unit test `TestUncompressedHelmTarError` (`files/tests/sanitization_test.go`) that builds an in-memory uncompressed tar containing `Chart.yaml` and asserts the new error path.

Files touched
- `files/error.go` — new error constructor for uncompressed `.tar`.
- `files/identification.go` — improved detection and error handling around `loader.LoadArchive`.
- `files/tests/sanitization_test.go` — added unit test.

Why
- Helm expects charts as gzipped tarballs (e.g., `.tgz`). When an uncompressed `.tar` is supplied the gzip reader fails with an opaque "gzip: invalid header". This change makes the failure actionable for users.

Tests
- Added unit test covering the uncompressed `.tar` case.
- Ran `go test ./...` locally; tests pass.

Related
- Fixes: #813

How to validate locally
1. Run the test suite:
   ```bash
   go test ./...

lekaf974 commented on files/error.go:
```diff
@@ -260,6 +260,29 @@ func ErrInvalidHelmChart(fileName string, err error) error {
 
+// ErrUncompressedTarProvided returns an error explaining that Helm requires
+// compressed tarballs (tgz / tar.gz) and that an uncompressed .tar was provided.
+func ErrUncompressedTarProvided(fileName string, err error) error {
```
```suggestion
func ErrUncompressedTar(fileName string, err error) error {
```

lekaf974 commented on files/error.go:
```diff
@@ -260,6 +260,29 @@ func ErrInvalidHelmChart(fileName string, err error) error {
+		"Create the Helm chart archive using 'helm package' or gzip the archive before uploading.",
+	}
+
+	return errors.New(ErrInvalidHelmChartCode, errors.Critical, sdescription, ldescription, probableCause, remedy)
```
We are using values directly without variables in errors usually, could we keep implementation similar here ?

bashSunny101 commented:
I have made requested changes to this PR, kindly review it.
@lekaf974

lekaf974 commented on files/error.go:
```diff
@@ -260,27 +261,20 @@ func ErrInvalidHelmChart(fileName string, err error) error {
+		ErrUncompressedTarCode,
+		errors.Critical,
+		[]string{fmt.Sprintf("The file '%s' appears to be an uncompressed TAR archive.", fileName)},
+		[]string{fmt.Sprintf("Helm expects chart archives to be compressed (e.g., .tgz or .tar.gz). Error details: %s", err.Error())},
```
```suggestion
		[]string{fmt.Sprintf("Expected archives to be compressed (e.g., .tgz or .tar.gz). Error details: %s", err.Error())},
```

lekaf974 commented on files/error.go:
```diff
@@ -260,27 +261,20 @@ func ErrInvalidHelmChart(fileName string, err error) error {
+		[]string{"The archive was created as an uncompressed .tar, but Helm requires compressed archives."},
+		[]string{
+			"Compress the .tar file to .tgz or .tar.gz and try again.",
+			"Create the Helm chart archive using 'helm package' or gzip the archive before uploading.",
```
```suggestion
			"If it is a Helm chart archive, create using 'helm package' or gzip the archive before uploading.",
```

lekaf974 reviewed (commented):
another review after we will be good to merged

Good job

lekaf974 commented:
@bashSunny101 do you have time to do last changes so it can be merged ?

bashSunny101 commented:
@lekaf974  
Done, Let me know if any further changes are required.

bashSunny101 commented:
@lekaf974 I've addressed the requested changes:

- Added `github.com/stretchr/testify/assert` import to the test file
- Replaced the `if err == nil` check with `assert.NotNil(t, err)` 
- Cleaned up the duplicate error checking code

All changes have been committed with sign-off and pushed. Please let me know if there's anything else that needs to be updated. Thanks for the review!
